fvwm/ewmh_events.c

```diff
@@ -467,6 +467,60 @@ int ewmh_MoveResize(
 	return 0;
 }
 
+int ewmh_WMFullscreenMonitors(
+	FvwmWindow *fw, XEvent *ev, window_style *style, unsigned long any)
+{
+	int top, bottom, left, right, num_screens;
+	Bool had_fsmonitors_set = fw->fullscreen_monitors_set;
+
+	if (ev == NULL)
+	{
+		return 0;
+	}
+
+	if ((num_screens = FScreenNumberOfScreens()) == 0)
+	{
+		/* no Xinerama (or disabled) */
+		return 0;
+	}
+
+	fw->fullscreen_monitors_set = False;
+
+	top = (int)ev->xclient.data.l[0];
+	bottom = (int)ev->xclient.data.l[1];
+	left = (int)ev->xclient.data.l[2];
+	right = (int)ev->xclient.data.l[3];
+	if (top < num_screens ||
+	    bottom < num_screens ||
+	    left < num_screens ||
+	    right < num_screens)
+	{
+		fw->fullscreen_monitors_set = True;
+	}
+
+	if (fw->fullscreen_monitors_set)
+	{
+		long data[4];
+		data[0] = top;
+		data[1] = bottom;
+		data[2] = left;
+		data[3] = right;
+
+		ewmh_ChangeProperty(FW_W(fw),
+			"_NET_WM_FULLSCREEN_MONITORS",
+			EWMH_ATOM_LIST_CLIENT_WIN,
+			(unsigned char *)data, 4);
+	}
+	else if (had_fsmonitors_set)
```

Comment:
I basically copied the logic from Compiz.

Anyway, the main purpose of the overall change was I'd need it right now, thus need it for fvwm 2.x.

I don't see _NET_WM_FULLSCREEN_MONITORS in fvwm3 codebase yet. Do you want me to suggest an implementation for it, rewritten based on your comments?

Reply:
No, FVWM 3.x doesn't exist yet.  The point I'm making is that there's no more significant code changes going into FVWM 2.x any more.  This change is one example of where the line is drawn.  AFAIAC, anything which attempts to use Xinerama is buggy, and undefined, as the protocol itself is all but dead, to say nothing of dwindling client support for it.

Reply:
Nevertheless, we have to cope with clients like VMware that insist on using the established standards (and from their point, who are p**ed off by having to constantly shoot moving targets).

Btw., it would be good if you updated the developers documentation:

http://fvwm.org/documentation/developers/

telling that any non-bugfix code changes for fvwm 2.x are not acceptable anymore.